### PR TITLE
Fix elasticsearch recipe

### PR DESCRIPTION
**Pull request by xvello**
- bind to _eth0_,_local_ to allow access out of container
- rename auto_conf file to overwrite standard one
- lower mem limit to 600 MB from 1 GB
- remove unused host port binding and volumes
- remove container_name, keep autogenerated ones

### CODE REVIEW

File: elasticsearch/standalone.compose [hunk 1]
```diff
@@ -4,24 +4,19 @@ services:
...
+      - "ES_JAVA_OPTS=-Xms256m -Xmx256m"
     ulimits:
       memlock:
         soft: -1
         hard: -1
-    mem_limit: 1g
-    volumes:
-      - esdata:/usr/share/elasticsearch/data
-    ports:
-      - 9200:9200
```
Comment (xvello) [inline elasticsearch/standalone.compose:21]: The second value creates a host port binding (meaning you can access it via localhost:9200 on the host), that is unnecessary and prevents several instances to run

File: elasticsearch/standalone.compose [hunk 2]
```diff
@@ -4,24 +4,19 @@ services:
...
-      - network.host=__site__
-      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+      - network.host=_eth0_,_local_
+      - "ES_JAVA_OPTS=-Xms256m -Xmx256m"
     ulimits:
       memlock:
         soft: -1
         hard: -1
-    mem_limit: 1g
-    volumes:
```
Comment (xvello) [inline elasticsearch/standalone.compose:18]: As we don't want to keep the data, no need to mount a volume.
Furthermore, these dangling volumes were a pain when restarting the containers

**Review by eplanet — APPROVED** [review]
It works perfectly well!

The only remaining gotcha with ES, is this `max_map_count` parameter. It is well documented, but is it sufficient?

**Comment by xvello** [comment]
Indeed, I felt into it ;)
Compose allows to set some `sysctls`, but only the ones that are settable for a cgroup. This one is not supported :(
We'd need to implement a pre-start script to automate that, but then, should it `sudo` or `su`? I can add (see README.md) to both flavor descriptions for the time being

**Pull request merged**